src/common/transformations/src/transformations/resolve_names_collisions.cpp

```diff
@@ -43,19 +43,38 @@ bool ov::pass::ResolveNameCollisions::run_on_model(const std::shared_ptr<ov::Mod
             continue;
         }
 
-        int64_t cnt = 2;
-        for (const auto& op : same_named_ops) {
-            // check if op has OV autogenerated friendly name. unique and friendly names have to be equal.
-            bool is_autogenerated = op->m_friendly_name.empty();
-            if (!is_autogenerated) {
-                continue;
-            }
+        auto update_name = [&](ov::Node* op, int64_t& cnt) {
             // add a prefix "_counter" to the autogenerated name to make it unique.
-            auto new_name = op->get_friendly_name() + "_" + std::to_string(cnt++);
+            const auto& old_name = op->get_friendly_name();
+            auto new_name = old_name + "_" + std::to_string(cnt++);
             while (visited_nodes.find(new_name) != visited_nodes.end()) {
-                new_name = op->get_friendly_name() + "_" + std::to_string(cnt++);
+                new_name = old_name + "_" + std::to_string(cnt++);
             }
             op->set_friendly_name(new_name);
+        };
+
+        if (m_resolve_all_names) {
+            int64_t cnt = 0;
+            for (const auto& op : same_named_ops) {
+                // we don't need to change name of the first op in this list.
+                // the expected result: Name, Name_1, Name_2 etc
+                if (cnt == 0) {
+                    cnt++;
+                    continue;
+                }
+                update_name(op, cnt);
+            }
+        } else {
+            int64_t cnt = 2;
```

Comment:
maybe cnt = 1 is better

Reply:
Looks strange that you are using different initial value for counter. Also, I think you can use the same logic for both cases `m_resolve_all_names` that wiil determine if check `op->m_friendly_name.empty()`

Reply:
I will resolve it in the next PR